pkg/ring/kv/consul/client_test.go

```diff
@@ -0,0 +1,136 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	consul "github.com/hashicorp/consul/api"
+
+	"github.com/cortexproject/cortex/pkg/ring/kv/codec"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type stringCodec struct{}
+
+func (c stringCodec) Encode(d interface{}) ([]byte, error) {
+	if d == nil {
+		return nil, fmt.Errorf("nil")
+	}
+	s, ok := d.(string)
+	if !ok {
+		return nil, fmt.Errorf("not string: %T", d)
+	}
+
+	return []byte(s), nil
+}
+
+func (c stringCodec) Decode(d []byte) (interface{}, error) {
+	return string(d), nil
+}
+
+var _ codec.Codec = &stringCodec{}
```

Comment:
What is this doing?

Reply:
By adding that line, IntelliJ then offers me to add missing methods from the interface. I forgot to remove it afterwards.